### ISSUE TITLE
Example using user last_login

### ISSUE THREAD

**Issue by wadewilliams** [ra-systems/django-slick-reporting#15]
As a proof of concept, I'd like to see a chart for daily users who logged in. I can get a report of users who logged in during a month easily enough, but displaying a chart is illusive based off the docs. The example in the readme doesn't match the example in the documentation. It'd be nice to have an example with something built-in like user `last_login` since that's universally available for Django users, without having to understand the context of a 3rd party app example with "orders" 

Something like 

```py
from slick_reporting.views import SlickReportView
from django.contrib.auth import get_user_model

User = get_user_model()


class LogIns(SlickReportView):
    report_model = User
    date_field = 'last_login'

    # The columns you want to display
    columns = ['last_login']  # I'm sure this is wrong?  I want to group by day?

    time_series_pattern = 'daily'
    time_series_columns = ['__total_quantity__']  # not clear if this is a keyword that slick reporting understands or something else

    # Charts
    chart_settings = [{
        'type': 'column',
        'data_source': ['sum__value'],  # definitely wrong but not sure how to fix
        'plot_total': False,
        'title_source': 'last_login',  # seems wrong
        'title': _('Detailed Columns'),  # just an error not sure what this is 

    }, ]
```

**Comment by RamezIssac** [comment]
Hello @wadewilliams 

Thank you for your review.

Documenting by something all Django users can relate to is a brilliant idea.
Also, yes the documentation need way more examples.

Here are my thoughts reading your `LogIns` example

Grouping by date (or field types other then ForeignKeys) is kinda tricky / not fully implemented .. up to the current version 0.4.2 you can only group by Foreign Keys. i see this feature importance and I will certainly work on it.


`__total_quantity__` is a ReportFIeld that's already added in slick reporting.
It stand for "Sum of a field called 'quantity'".. You can see its definition here https://github.com/ra-systems/django-slick-reporting/blob/7751f7d8d83bfadf38b4b0870fd6ba8a254fe995/slick_reporting/fields.py#L362
in this line https://github.com/ra-systems/django-slick-reporting/blob/7751f7d8d83bfadf38b4b0870fd6ba8a254fe995/slick_reporting/fields.py#L369 it's added  to the SlickReporting fields registry so you can refer to it by name in ``SlickReportingView`` columns

For ``chart_settings``:
`data_source`: is the name of the field that contains the values we want to plot. 
while `title_source` is the name of the field that contains the respective labels


To achieve a report grouped by day showing the number of user last_login on that day,  i would look something like this

```python
from django.db.models import Count

# let's create Report Field first 

class CountLogin(SlickReportField):
    name = 'count__login' # a unique name
    verbose_name = _('Login Count')
    calculation_field = 'last_login'
    calculation_method = Count 
    is_summable = True

field_registry.register(CountLogin)


# Now our view
class LogIns(SlickReportView):
    report_model = User
    date_field = 'last_login'

    group_by = 'last_login`

    # The columns you want to display
    columns = ['last_login', 'count__login'] 

    time_series_pattern = 'daily'
    time_series_columns = ['count__login']

    # Charts
    chart_settings = [{
        'type': 'column',
        'data_source': ['count__login'], # the name of the field that contains the values we want to plot
        'title_source': 'last_login',  # and that the labels'
        'plot_total': False,
        'title': _('Logins Per Day'),  #  you can share any errors stack trace, i'd be happy to look at them.

    }, ]
```

This will output something like this 

Date | login count | login count day 1 | login count day 2 | login count day 3 |  ..  | login count day n |

**Comment by RamezIssac** [comment]
On a second thought :

> As a proof of concept, I'd like to see a chart for daily users who logged in

Here using the `last_login` date field is not an accurate measure.. as a user who will login another day will be removed from the day before.
To achieve what you want probably you'd need a model where you record a user login only once per day.

However, 
we still can use the User table to know how many users _joined_ per day .

**Comment by wadewilliams** [comment]
Thanks for the quick turn around!

> Here using the last_login date field is not an accurate measure.. as a user who will login another day will be removed from the day before.

💯 fair enough - Was more just trying to pull something that is universally approachable. 

As far as the code goes, here's what I've got: 

```
from django.db.models import Count
from django.contrib.auth import get_user_model

from slick_reporting.fields import SlickReportField
from slick_reporting.registry import field_registry
from slick_reporting.views import SlickReportView

User = get_user_model()


class CountLogin(SlickReportField):
    name = 'count__logins' # a unique name
    verbose_name = 'Login Count'
    calculation_field = 'last_login'
    calculation_method = Count
    is_summable = True

field_registry.register(CountLogin)


# Now our view
class LogIns(SlickReportView):
    report_model = User
    date_field = 'last_login'

    group_by = 'last_login'

    # The columns you want to display
    columns = ['last_login', 'count__logins']

    time_series_pattern = 'daily'
    time_series_columns = ['count__logins']

    # Charts
    chart_settings = [{
        'type': 'column',
        'data_source': ['count__logins'], # the name of the field that contains the values we want to plot
        'title_source': 'last_login',  # and that the labels'
        'plot_total': False,
        'title': 'Logins Per Day' #  you can share any errors stack trace, i'd be happy to look at them.

    }, ]
```

Stack trace: 
```
...
line 18, in <module>
    field_registry.register(CountLogin)
File "/root/.local/share/virtualenvs/app-4PlAip0Q/lib/python3.8/site-packages/slick_reporting/registry.py", line 19, in register
    raise AlreadyRegistered(f'The field name {report_field.name} is used before and `override` is False')
django.contrib.admin.sites.AlreadyRegistered: The field name count__logins is used before and `override` is False
```

If I comment out `field_registry.register(CountLogin)` since it appears that it's already registered somehow, I get this error:

```
...
line 22, in <module>
    class LogIns(SlickReportView):
File "/root/.local/share/virtualenvs/app-4PlAip0Q/lib/python3.8/site-packages/slick_reporting/views.py", line 238, in __init_subclass__
    cls.report_generator_class.check_columns(cls.columns, cls.group_by, cls.report_model)
  File "/root/.local/share/virtualenvs/app-4PlAip0Q/lib/python3.8/site-packages/slick_reporting/generator.py", line 430, in check_columns
    field = model_to_use._meta.get_field(col)
AttributeError: 'NoneType' object has no attribute '_meta'
```

Thanks for your assistance.

**Comment by RamezIssac** [comment]
Yeah, Grouping on dates in not yet supported. However i just  started it :)

You can check the progresshere -> https://github.com/ra-systems/django-slick-reporting/tree/feature/group_by_date_support

**Comment by RamezIssac** [comment]
Using the branch feature/group_by_date_support i was able to get this example working 

```python
# fields.py,
# a new file where we create the Report fields to avoid any circular loading or errors like AlreadyRegistered

from django.db.models import Count
from slick_reporting.fields import SlickReportField
from slick_reporting.registry import field_registry
from django.utils.translation import ugettext_lazy as _

field_registry.register(SlickReportField.create(Count, 'id', 'count__logins', verbose_name=_('Count')))
```

Then 

```python
# apps.py
# we load the fields.py so the fields are registered and available

from django.apps import AppConfig


class AppNameConfig(AppConfig):
    name = 'your_app_name' 

    def ready(self):
        super().ready()
        from . import fields

```

Finally the view.py

```python
class LogIns(SlickReportView):
    report_model = User
    date_field = 'date_joined'

    group_by = 'date_joined'

    columns = ['date_joined', 'count__logins']
    
    chart_settings = [{
        'type': 'bar',
        'data_source': ['count__logins'],
        'title_source': 'date_joined',
        'plot_total': False,
        'title': 'Logins Per Day'

    }, ]
```

**Comment by RamezIssac** [comment]
Hello @wadewilliams 

Today a new released is pushed to pypi where you can group by date field.
Also, and inspired by our example conversation, i created this site https://django-slick-reporting.com/ to demo aspects of Django-Slick-Reporting... Still work in progress though :) 

Wish you a great day !
